spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -159,6 +161,8 @@
 
 	private Set<PosixFilePermission> permissions;
 
+	private volatile BiConsumer<File, Message<?>> newFileCallback;
```

Comment:
We don't use `volatile` for setter's properties.
It make sense only for those properties which can be modified from different thread, e.g. in the `start()`/`stop()` callback.
But not for setters, please.

Reply:
I just copied `volatile` since all non-final properties except `permissions` are `volatile` as well, didn't know, will fix.